### DISCUSSION TITLE
Windows Web Services API vs CURL

### DISCUSSION THREAD

**Discussion by hiic** [curl/curl#13414] [General]
I have to replace existing library that communicates with the soap-based web service on windows platform only. The library uses Windows Web Services API to build and send soap messages over the network. The new library must be cross-platform (Windows and Linux). Few of our existing projects already use CURL. My question is whether WWSAPI has advantage over CURL with respect to client/server communication, in terms of resource efficiency, speed, reliability? Perhaps, I need to consider using WWSAPI on windows and CURL on Linux. I want to avoid degrading the existing functionality that we currently use on Windows by switching to CURL. This may not be the case but I do not know. I understand the question may be too broad.

**Comment by bagder** [comment]
> My question is whether WWSAPI has advantage over CURL with respect to client/server communication, in terms of resource efficiency, speed, reliability? 

I would guess that curl is faster and more reliable, but I don't know. I would urge you to set up some basic benchmarks and measure your own use case. I would love to see some numbers.

If there would be *any* issues, since curl is open source and actively developed, they could be sorted out quickly - which nothing from Microsoft can offer. You can do it yourself, get help by the community or even purchase support.

**Comment by jay** [comment]
Use the same transfer library on both Windows and Linux. Less maintenance. IMO whether it's 0.04 seconds faster or slower should take a backseat to that.

**Reply by hiic** [reply]
Thank you Jay and Bagder, appreciate your reply. Yes, ideally should use same library for both platforms and yes a small time difference should not matter that much. The reliability is more important. It will take some time but once available, I will share the timings.

**Reply by gvanem** [reply]
WWSAPI is yet another dead API by Microsoft (powerful but a clumsy XML-based API).
Newest question for it on [Stackoverflow](https://stackoverflow.com/search?tab=newest&q=WWSAPI&searchOn=3) is 4 years old.

**Reply by hiic** [reply]
Thanks. For the xml part it is convenient as it wraps the soap xml serialize/deserialize from/to objects but I  do not see this as a big deal, can  implement functions to work with xml for several WSDL API, not a big concern for me.

**Reply by jay** [reply]
> can implement functions to work with xml for several WSDL API, not a big concern for me.

Depending on the complexity you may want to consider a SOAP library and the one you choose may have its own transfer capability. gSOAP (GPL or commercial) for example has its own and a curl plugin option.
